src/hotspot/share/oops/symbolHandle.hpp

```diff
@@ -76,6 +90,19 @@ class SymbolHandleBase : public StackObj {
     Symbol::maybe_decrement_refcount(_temp);
   }
 
+  // Keep this symbol alive for some time to allow for reuse.
+  // Temp symbols for the same string can often be created in quick succession,
+  // and this queue allows them to be reused instead of churning.
+  void add_to_cleanup_delay_queue(Symbol* sym) {
+    if (!_cleanup_delay_enabled) return;
+    sym->increment_refcount();
+    uint i = Atomic::add(&_cleanup_delay_index, 1u) % CLEANUP_DELAY_MAX_ENTRIES;
```

Comment:
When this wraps around, it will skip a few values - but I don't see a problem with that.

```
x = 4294967293, x % 100 = 93
x = 4294967294, x % 100 = 94
x = 4294967295, x % 100 = 95
x = 0, x % 100 = 0
x = 1, x % 100 = 1
x = 2, x % 100 = 2
```

Reply:
I don't see a problem either.  I wish I were better with integers.  I was wondering if the max entries should be a power of two to make the % operation faster.

Reply:
Good idea, changed to 128. I'm no expert but the code looks better for it here: https://godbolt.org/z/7cYv9Pzz7.

Reply:
I like it!

Reply:
A common thing is to do the mod conversion right here, without relying on optimizing compilers:
`% CLEANUP_DELAY_MAX_ENTRIES` -> `& (CLEANUP_DELAY_MAX_ENTRIES - 1))`.

Reply:
If you do `constexpr` for the constant, then you can also do `STATIC_ASSERT(is_power_of_2(CLEANUP_DELAY_MAX_ENTRIES))`.